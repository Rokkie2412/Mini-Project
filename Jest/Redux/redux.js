export const ADD = 'ADD'
export const DELETE = 'DELETE'

export const addArray = (data) =>({
    type:ADD,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch(action.type){
        case ADD:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE:
            return{
                ...state,arr:state.arr.filter(arr => arr.id !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer