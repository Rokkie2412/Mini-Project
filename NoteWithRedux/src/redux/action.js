import { ADDARRAY } from "./actionType";

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})