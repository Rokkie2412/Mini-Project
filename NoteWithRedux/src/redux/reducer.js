import { ADDARRAY } from "./actionType";

const initialState = {
    arr:[],
    
};

export const mainReducer = (state = initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
           
        default:
            return state;
    }
}