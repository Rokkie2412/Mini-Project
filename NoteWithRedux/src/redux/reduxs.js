import remove from "lodash.remove"
export const ADDARRAY = 'ADDARRAY'
export const DELETENOTE = 'DELETENOTE'
export const EDITNOTE = 'EDITNOTE'

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})

export const deleteFromArray = (title) => ({
    type:DELETENOTE,
    payload:title
})
export const EditNote = (title) => ({
    type:EDITNOTE,
    payload:title,
})



const initialState = []

export const mainReducer = (state=initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return[
                ...state,action.payload
            ]
        case DELETENOTE:
            const deleteNewArray = remove(state, obj => {
                return obj.title != action.payload
            })
            return deleteNewArray
        case EDITNOTE:
            // const filterDelete = map(state,obj =>{
            //     return obj.title == action.payload
            // })
            const filtered = state.filter((state,obj)=>{
                if(obj.title === action.payload){
                    var tamp = obj.title
                    console.log(tamp);
                }
            })
        default:
            return state;
    }
}

export default mainReducer