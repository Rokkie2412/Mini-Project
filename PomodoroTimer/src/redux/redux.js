
/**
 * @format
 * @flow
 */
export const EXCHAHNGETOSECONDS : string = 'EXCHANGETOSECOND'
export const EXCHAHNGETOSECONDBREAK :string = 'EXCHANGETOSECONDBREAK'
export const EXCHAHNGETOSECONDLONGBREAK :string = 'EXCHANGETOSECONDLONGBREAK'
export const GETDATA:string = 'GETDATA'

export const exchangetosecwork = (hour:string,minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDS,
    payloadhourwork : hour,
    payloadminutework : minute,
    payloadsecondwork : second
})

export const exchangetosecBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDBREAK,
    payloadminutebreak : minute,
    payloadsecondbreak : second
})

export const exchangetosecLongBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDLONGBREAK,
    payloadminutelongbreak : minute,
    payloadsecondlongbreak : second
})

let initialState : Object  ={
    resultWorkinSecond : 0,
    resultBreakinSecond : 0,
    resultLongBreakinSecond : 0
}

export const mainReducer = (state:Object=initialState, action:function):Function =>{
    switch (action.type) {
        case EXCHAHNGETOSECONDS:
            
            const hourTosecond : number = action.payloadhourwork * 3600
            const minuteTosecond : number = action.payloadminutework * 60
            const res : number = hourTosecond+minuteTosecond+action.payloadsecondwork
            return {
                ...state,
                resultWorkinSecond:res
            }
        case EXCHAHNGETOSECONDBREAK:
            const minuteTosecondBreak : number= action.payloadminutebreak * 60
            const resBreak : number = minuteTosecondBreak+action.payloadsecondbreak
            return {
                ...state,
                resultBreakinSecond:resBreak
            }
        case EXCHAHNGETOSECONDLONGBREAK:
            const minuteToSecondLongBreak : number= action.payloadminutelongbreak * 60
            const resLongBreak : number = minuteToSecondLongBreak+action.payloadsecondlongbreak
            return{
                ...state,
                resultLongBreakinSecond:resLongBreak
            }
        default:
            return state
    }
}

export default mainReducer