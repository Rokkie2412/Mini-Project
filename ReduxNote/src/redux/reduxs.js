import remove from "lodash.remove";
export const ADD_NOTE = 'ADDNOTE'
export const DELETE_NOTE = 'DELETE_NOTE'

export const addArray = (data) =>({
    type:ADD_NOTE,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE_NOTE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch (action.type) {
        case ADD_NOTE:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE_NOTE:
            // const deleteNewArray = remove(state=initialState,obj =>{
            //     return obj.title != action.payload
            // })
            // return state.arr = deleteNewArray
            return{
                ...state,arr:state.arr.filter(arr => arr.title !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer