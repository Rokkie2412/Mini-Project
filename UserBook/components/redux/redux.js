export const ADD_CONTACT = 'ADD_CONTACT'
export const SHOW_CONTACT = 'SHOW_CONTACT'
export const EDITCONTACT = 'EDITCONTACT'

const ApiUrl = 'https://simple-contact-crud.herokuapp.com/contact'


export const addContact = (firstName,lastName,age,image) =>({
    type:ADD_CONTACT,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

export const editContact = (itemId,firstName,lastName,age,image) => ({
    type:EDITCONTACT,
    payloadID : itemId,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

let initialState = {
    arr:[],
}

export const mainReducer = (state=initialState, action) => {
    switch (action.type) {
        case ADD_CONTACT:
            const postContact = () =>{
                fetch('https://simple-contact-crud.herokuapp.com/contact',{
                    method:'POST',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(postContact())
        case EDITCONTACT :
            const editContact = () =>{
                const tempid = action.payloadID 
                
                fetch(`https://simple-contact-crud.herokuapp.com/contact/${tempid}`,{
                    method:'PUT',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                 }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(editContact())
        default:
            return state
    }
}