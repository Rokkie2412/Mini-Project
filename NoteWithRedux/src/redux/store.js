import { createStore } from "redux";
import {mainReducer} from "./reduxs";

export const store = createStore(mainReducer)