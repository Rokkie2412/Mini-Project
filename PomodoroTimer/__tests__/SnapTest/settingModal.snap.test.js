import React from 'react';
import {TextInput} from 'react-native'
import {create,act,fireEvent} from 'react-test-renderer';
import SettingsModal from '../../src/components/SetttingsModal';;
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {mainReducer} from '../../src/redux/redux';
import { SearchBar } from 'react-native-screens';
import {shallow, mount, render} from 'enzyme'

jest.mock('react-native-vector-icons/Ionicons', () => 
jest.requireActual('react-native-vector-icons/Ionicons'))
jest.mock('react-native-background-timer', () => 'react-native-background-timer')
jest.mock('react-native-push-notification', () => 'react-native-push-notification')
const func = jest.fn()

const store = createStore(mainReducer, {status: 'default status'});
const tree = create(
  <Provider store={store}>
    <SettingsModal setshow={func}/>
  </Provider>,
);

describe('Render Setting Modal Screen', () => {
  describe('Snapshot for Setting',()=>{
    it('Should render correctly for setting modal',()=>{
      expect(tree).toMatchSnapshot();
    })
  })
  describe('Test Button for close modal',()=>{
    it('Should be acted to close by change setter',()=>{
      const cancleButton = tree.root.findByProps({testID:'CloseSetting'}).props
      act(()=>cancleButton.onPress())
    })
  })
  describe('Test Button for Save setting and close modal',()=>{
    it('Should be acted to close by change setter',()=>{
      const SaveButton = tree.root.findByProps({testID:'SaveSetting'}).props
      act(()=>SaveButton.onPress())
    })
  })
  describe('Text Input calling',()=>{
    it('Should be acted to set hour',()=>{
      const sethour = tree.root.findByProps({testID:'sethour'}).props
      act(()=>sethour.onChangeText())
    })
    it('Should be acted to set minute',()=>{
      const setminute = tree.root.findByProps({testID:'setminute'}).props
      act(()=>setminute.onChangeText())
    })
    it('Should be acted to set second',()=>{
      const setsecond = tree.root.findByProps({testID:'setsecond'}).props
      act(()=>setsecond.onChangeText())
    })
    it('Should be acted to set second',()=>{
      const setshortsecond = tree.root.findByProps({testID:'setshortsecond'}).props
      act(()=>setshortsecond.onChangeText())
    })
    it('Should be acted to set minute',()=>{
      const setshortminute = tree.root.findByProps({testID:'setshortminute'}).props
      act(()=>setshortminute.onChangeText())
    })
    it('Should be acted to set minute',()=>{
      const setlongminute = tree.root.findByProps({testID:'setlongminute'}).props
      act(()=>setlongminute.onChangeText())
    })
    it('Should be acted to set minute',()=>{
      const setlongsecond = tree.root.findByProps({testID:'setlongsecond'}).props
      act(()=>setlongsecond.onChangeText())
    })
  })
})

