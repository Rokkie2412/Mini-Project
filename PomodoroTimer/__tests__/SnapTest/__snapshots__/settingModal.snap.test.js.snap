// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Setting Modal Screen Snapshot for Setting Should render correctly for setting modal 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "backgroundColor": "#393E46",
        "borderRadius": 30,
        "flex": 1,
        "height": 933.8,
        "margin": "auto",
        "marginHorizontal": "5%",
        "marginVertical": "27%",
        "width": 675,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EEEEEE",
            "fontSize": 22,
            "fontWeight": "bold",
            "margin": 10,
          }
        }
      >
        Settings
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#222831",
          "borderRadius": 20,
          "marginHorizontal": 20,
          "marginVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EEEEEE",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 7,
            "textAlign": "center",
          }
        }
      >
        Set Work Timer
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          keyboardType="number-pad"
          onChangeText={[Function]}
          placeholder="00"
          placeholderTextColor="#EEEEEE"
          style={
            Object {
              "borderBottomColor": "#eeeeee",
              "borderBottomWidth": 2,
              "color": "#EEEEEE",
              "fontSize": 18,
              "margin": 10,
              "marginHorizontal": 8,
              "padding": 10,
              "textAlign": "center",
            }
          }
          testID="sethour"
          value=""
        />
        <TextInput
          keyboardType="number-pad"
          onChangeText={[Function]}
          placeholder="00"
          placeholderTextColor="#EEEEEE"
          style={
            Object {
              "borderBottomColor": "#eeeeee",
              "borderBottomWidth": 2,
              "color": "#EEEEEE",
              "fontSize": 18,
              "margin": 10,
              "marginHorizontal": 8,
              "padding": 10,
              "textAlign": "center",
            }
          }
          testID="setminute"
          value=""
        />
        <TextInput
          keyboardType="number-pad"
          onChangeText={[Function]}
          placeholder="00"
          placeholderTextColor="#EEEEEE"
          style={
            Object {
              "borderBottomColor": "#eeeeee",
              "borderBottomWidth": 2,
              "color": "#EEEEEE",
              "fontSize": 18,
              "margin": 10,
              "marginHorizontal": 8,
              "padding": 10,
              "textAlign": "center",
            }
          }
          testID="setsecond"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#222831",
          "borderRadius": 20,
          "marginHorizontal": 20,
          "marginVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EEEEEE",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 7,
            "textAlign": "center",
          }
        }
      >
        Set Short Break Timer
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          keyboardType="number-pad"
          onChangeText={[Function]}
          placeholder="00"
          placeholderTextColor="#EEEEEE"
          style={
            Object {
              "borderBottomColor": "#eeeeee",
              "borderBottomWidth": 2,
              "color": "#EEEEEE",
              "fontSize": 18,
              "margin": 10,
              "marginHorizontal": 8,
              "padding": 10,
              "textAlign": "center",
            }
          }
          testID="setshortminute"
          value=""
        />
        <TextInput
          keyboardType="number-pad"
          onChangeText={[Function]}
          placeholder="00"
          placeholderTextColor="#EEEEEE"
          style={
            Object {
              "borderBottomColor": "#eeeeee",
              "borderBottomWidth": 2,
              "color": "#EEEEEE",
              "fontSize": 18,
              "margin": 10,
              "marginHorizontal": 8,
              "padding": 10,
              "textAlign": "center",
            }
          }
          testID="setshortsecond"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#222831",
          "borderRadius": 20,
          "marginHorizontal": 20,
          "marginVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EEEEEE",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 7,
            "textAlign": "center",
          }
        }
      >
        Set Long Break Timer
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          keyboardType="number-pad"
          onChangeText={[Function]}
          placeholder="00"
          placeholderTextColor="#EEEEEE"
          style={
            Object {
              "borderBottomColor": "#eeeeee",
              "borderBottomWidth": 2,
              "color": "#EEEEEE",
              "fontSize": 18,
              "margin": 10,
              "marginHorizontal": 8,
              "padding": 10,
              "textAlign": "center",
            }
          }
          testID="setlongminute"
          value=""
        />
        <TextInput
          keyboardType="number-pad"
          onChangeText={[Function]}
          placeholder="00"
          placeholderTextColor="#EEEEEE"
          style={
            Object {
              "borderBottomColor": "#eeeeee",
              "borderBottomWidth": 2,
              "color": "#EEEEEE",
              "fontSize": 18,
              "margin": 10,
              "marginHorizontal": 8,
              "padding": 10,
              "textAlign": "center",
            }
          }
          testID="setlongsecond"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="CloseSetting"
      >
        <Text
          style={
            Object {
              "backgroundColor": "#00ADB5",
              "borderRadius": 50,
              "color": "#EEEEEE",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": 30,
              "padding": 15,
            }
          }
        >
          Cancel
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="SaveSetting"
      >
        <Text
          style={
            Object {
              "backgroundColor": "#00ADB5",
              "borderRadius": 50,
              "color": "#EEEEEE",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": 30,
              "padding": 15,
            }
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
